libtenzir/builtins/contexts/bloom_filter.cpp

```
@@ -0,0 +1,240 @@
+    // chunks until the upstream pipeline is complete. What we need in the
+    // context API is a signal that we're done.
+    return ec::unimplemented;
+  }
+
+  auto update(context::parameter_map) -> caf::expected<update_result> override {
+    return caf::make_error(ec::unimplemented,
+                           "bloom-filter context can not be updated with void");
+  }
+
+  auto save() const -> caf::expected<chunk_ptr> override {
+    std::vector<std::byte> buffer;
+    if (auto err = convert(bloom_filter_, buffer))
+      return add_context(err, "failed to serialize Bloom filter context");
+    return chunk::make(std::move(buffer));
+  }
+
+private:
+  dcso_bloom_filter bloom_filter_;
+};
+
+class plugin : public virtual context_plugin {
+  auto initialize(const record&, const record&) -> caf::error override {
+    return caf::none;
+  }
+
+  auto name() const -> std::string override {
+    return "bloom-filter";
+  }
+
+  auto make_context(context::parameter_map parameters) const
+    -> caf::expected<std::unique_ptr<context>> override {
+    auto capacity = parameters["capacity"];
+    if (not capacity)
+      return caf::make_error(ec::parse_error, "no --capacity provided");
+    auto fpp = parameters["fp-prob"];
+    if (not fpp)
+      return caf::make_error(ec::parse_error, "no --fp-prob provided");
+    auto n = uint64_t{0};
+    if (not parsers::u64(*capacity, n)) {
+      return caf::make_error(ec::invalid_argument,
+                             "--capacity is not an integer");
+    }
+    auto p = double{0.0};
+    if (not parsers::real(*fpp, p)) {
+      return caf::make_error(ec::invalid_argument, "--fp-prob is not a double");
+    }
+    if (n == 0) {
+      return caf::make_error(ec::invalid_argument, "--capacity must be > 0");
+    }
+    if (p < 0.0 || p > 1.0) {
+      return caf::make_error(ec::invalid_argument, "--fp-prob not in (0,1)");
+    }
+    return std::make_unique<bloom_filter_context>(n, p);
```

Please iterate over all key-value pairs in `parameters` so that you can return a meaningful error when an option is provided that may not be used with the bloom-filter context.

---

Solved in [15e245a](https://github.com/tenzir/tenzir/pull/3834/commits/15e245a58d6294ece98c1e20b55e85366f72f0c7).